deno/genadl.ts

```diff
@@ -14,37 +14,55 @@ async function main() {
 
   {
     //----------------------------------------------------------------------
-    // Generate typescript for the protoapp ui
+    // Generate typescript for the adl package
 
-    const outputDir = repo + "/ts/ui/src/adl-gen";
+    const outputDir = repo + "/ts/adl/src";
     await genTypescript({
       ...commonFlags,
-      adlModules: [
-        "protoapp.apis.ui",
-        "sys.adlast",
-        "common.ui",
-      ],
+      adlModules: ["protoapp.apis.ui", "sys.adlast", "common.ui"],
       tsStyle: "tsc",
       outputDir: outputDir,
       includeResolver: true,
       manifest: outputDir + "/.adl-manifest",
       generateTransitive: true,
       excludeAstAnnotations: [],
     });
+
+    // ensure pnpm dependencies are installed
+    let cmd = new Deno.Command("pnpm", {
+      args: ["install"],
+      cwd: repo + "/ts",
+    });
+    const installOutput = await cmd.output();
+    if (installOutput.stderr.length > 0) {
+      console.error(
+        "pnpm install stderr:",
+        new TextDecoder().decode(installOutput.stderr),
+      );
+    }
+
+    // build the adl package so it can be consumed
+    cmd = new Deno.Command("pnpm", {
+      args: ["run", "build"],
+      cwd: repo + "/ts/adl",
+    });
+    const buildOutput = await cmd.output();
+    if (buildOutput.stderr.length > 0) {
+      console.error(
+        "pnpm build stderr:",
+        new TextDecoder().decode(buildOutput.stderr),
+      );
+    }
```

Comment:
I think it's not the job of this script to build the ts/adl package. It doesn't build the rust adl package (or consume the adl derived code in any other way).